src/main/java/net/minecraftforge/fml/common/asm/transformers/SoundEngineFixTransformer.java

```diff
@@ -0,0 +1,137 @@
+package net.minecraftforge.fml.common.asm.transformers;
+
+import java.util.Iterator;
+
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.tree.AbstractInsnNode;
+import org.objectweb.asm.tree.ClassNode;
+import org.objectweb.asm.tree.FieldInsnNode;
+import org.objectweb.asm.tree.FieldNode;
+import org.objectweb.asm.tree.InsnNode;
+import org.objectweb.asm.tree.JumpInsnNode;
+import org.objectweb.asm.tree.LabelNode;
+import org.objectweb.asm.tree.MethodInsnNode;
+import org.objectweb.asm.tree.MethodNode;
+import org.objectweb.asm.tree.VarInsnNode;
+
+import net.minecraft.launchwrapper.IClassTransformer;
+
+public class SoundEngineFixTransformer implements IClassTransformer
+{
+    @Override
+    public byte[] transform(String name, String transformedName, byte[] basicClass)
+    {
+        if (transformedName.equals("paulscode.sound.Source"))
+        {
+            ClassNode classNode = new ClassNode();
+            ClassReader classReader = new ClassReader(basicClass);
+            classReader.accept(classNode, 0);
+
+            classNode.fields.add(new FieldNode(Opcodes.ACC_PUBLIC, "removed", "Z", null, null));
+
+            ClassWriter writer = new ClassWriter(ClassWriter.COMPUTE_MAXS);
+            classNode.accept(writer);
+            return writer.toByteArray();
+        }
+        else if (transformedName.equals("paulscode.sound.Library"))
+        {
+            ClassNode classNode = new ClassNode();
+            ClassReader classReader = new ClassReader(basicClass);
+            classReader.accept(classNode, 0);
+
+            MethodNode method = null;
+            for (MethodNode m : classNode.methods)
+            {
+                if (m.name.equals("removeSource") && m.desc.equals("(Ljava/lang/String;)V"))
+                {
+                    method = m;
+                    break;
+                }
+            }
+            if (method == null)
+                throw new RuntimeException("Error processing " + transformedName + " - no removeSource method found");
+
+            for (Iterator<AbstractInsnNode> iterator = method.instructions.iterator(); iterator.hasNext();)
```

Comment:
You are using an iterator on `method.instructions` and also modifying it in the loop without using the iterator. This is asking for CMEs, please find a way to do this loop without concurrent modification.

Reply:
Doesn't matter it will break the for loop after modifying it.

Reply:
Thats true, but it's still a issue waiting to happen. It's better to code defensively. Honestly you're just trying to find a position to do your modifications, so you can move the logic of changing the instructions out of the loop, and just using it to find the location.